projects/annotation-ui-lib/src/lib/data/annotation-store.service.ts

```diff
@@ -259,11 +259,16 @@ export class AnnotationStoreService implements OnDestroy {
             .then();
     }
 
-    deleteAnnotationById(annotationId: string) {
-        this.pdfAnnotateWrapper.getStoreAdapter()
-        .deleteAnnotation(this.pdfRenderService.getRenderOptions().documentId, annotationId)
-        .then(() => {
-            this.pdfRenderService.render();
+    deleteAnnotationById(annotationId: string, pageNumber: number) {
+        return new Promise(resolve => {
```

Comment:
This seems a bit odd... rather returning a new promise that resolves when the other promise has resolve why not just return that promise:

```
const renderOptions = this.pdfRenderService.getRenderOptions();

return this.pdfAnnotateWrapper.getStoreAdapter()
  .deleteAnnotation(renderOptions.documentId, annotationId)
  .then(() => {
              renderOptions.rotate = this.pdfRenderService.getPageRotation(renderOptions, {pageNumber: pageNumber});
              this.pdfAnnotateWrapper.renderPage(pageNumber, renderOptions);
              resolve(true);
            });
```

Also maybe think about extracting that inline function so it becomes:

```
const renderOptions = this.pdfRenderService.getRenderOptions();

return this.pdfAnnotateWrapper.getStoreAdapter()
  .deleteAnnotation(renderOptions.documentId, annotationId)
  .then(() => this.renderPage(renderOptions));
```

Reply:
Better yet... I see we're using `await` below... maybe use it here too.